packages/core/src/configuration/Configuration.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { Configuration, container } from '.';
+import { DefaultCredentialProvider } from './DefaultCredentialProvider';
+import { env } from 'process';
+
+describe('configuration', () => {
+  const testApi = 'test/api';
+  const testBus = 'test-bus';
+  const testToken = 'test-token';
+  let configuration: Configuration;
+
+  beforeAll(() => {
+    const credentialProvider = container.resolve(DefaultCredentialProvider);
+
+    configuration = new Configuration({
+      api: testApi,
+      bus: testBus,
+      credentials: {
+        token: testToken
+      },
+      credentialProviders: [credentialProvider]
+    });
+  });
+
+  it('should be a single instance', () => {
+    const configuration2 = container.resolve(Configuration);
+    expect(configuration).toBe(configuration2);
+  });
+
+  describe('get', () => {
+    it('should get api', () => {
+      const api = configuration.get('api');
+      expect(testApi).toEqual(api);
+    });
+
+    it('should get bus', () => {
+      const bus = configuration.get('bus');
+      expect(testBus).toEqual(bus);
+    });
+
+    it('should get credentials', () => {
+      const token = configuration.get('credentials')?.token;
+      expect(testToken).toEqual(token);
+    });
```

Comment:
You should test behavior and not data:

- should return a value by the key
- should return undefined if the key is not defined
- etc